src/nodes.coffee

```diff
@@ -1414,6 +1414,16 @@ exports.Value = class Value extends Base
         mergeLocationData @base.locationData, initialProperties[initialProperties.length - 1].locationData
     object
 
+  containsSoak: ->
```

Comment:
Based on playing around on AST Explorer, the idea seems to be that everything "to the right of" the initial soak in a property/call chain should use the `OptionalMemberExpression`/`OptionalCallExpression` AST node type

But then its own `optional` AST field is still `true` or `false` depending on whether *that step* of the chain is soaked or not